pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+# pylint: disable=too-many-arguments
+import copy
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+from pennylane import numpy as np
+
+
+def qsvt(A, angles, wires, convention=None):
+    r"""Implements the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ circuit using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+
+    Given a matrix :math:`A`, and a list of angles :math:`\vec{\phi}`, this template applies a
```

Comment:
Maybe good form to say from the start that this assumes block encoding is used via BlockEncode, which is mostly suited for simulators. Then link saying that the related operation QSVT allows users to input circuits implementing the block encoding